internal/types/document.go

```diff
@@ -306,6 +309,54 @@ func (d *Document) RemoveByPath(path Path) {
 	removeByPath(d, path)
 }
 
+// SetByPath sets value by given path.
```

Comment:
We should document what happens when path doesn't exist in the document.

Can we merge SetByPath and InsertByPath?

And let's add unit tests so we don't break that behavior

Reply:
Done.